File: algorithms/python/FindDuplicateFileinSystem/FindDuplicateFileinSystem.py
class Solution:
    def findDuplicate(self, paths: List[str]) -> List[List[str]]:
        d = defaultdict(list)
        for path in paths:
            items = path.split(' ')
            for i in range(1, len(items)):
                file, content = items[i].split('(')
                d[content[:-1]].append(items[0] + '/' + file)
        return [v for v in d.values() if len(v) > 1]

File: algorithms/js/FindDuplicateFileinSystem/FindDuplicateFileinSystem.js
/**
 * @param {string[]} paths
 * @return {string[][]}
 */
 var findDuplicate = function(paths) {
    const map = new Map();
    for (const path of paths) {
        let [p, ...files] = path.split(' ');
        for (const file of files) {
            let [name, content] = file.split(/\(|\)/);
            if (!map.has(content)) {
                map.set(content, []);
            }
            map.get(content).push(`${p}/${name}`);
        }
    }
    return [...map.values()].filter(item => item.length > 1);
};